### PR TITLE
feature: support the GridSearch strategy for hyperparameter optimization

### PR DESCRIPTION
**Pull request by repushko**
*Issue #, if available:* N/A

*Description of changes:* Now the AWS SagemakerHPO support new strategy: GridSearch. In this strategy, we asking the user don't specify the max_jobs value, because we'll recalculate it later. Because of that, in this CR the default value is None, but if the strategy is not a GridSearch -- we'll fallback to a previous default: 1. So, that's backward compatible

*Testing done:* expended unit tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/tuner.py [hunk 1]
```diff
@@ -258,7 +259,7 @@ def __init__(
                 evaluating training jobs. This value can be either 'Minimize' or
                 'Maximize' (default: 'Maximize').
             max_jobs (int or PipelineVariable): Maximum total number of training jobs to start for
-                the hyperparameter tuning job (default: 1).
+                the hyperparameter tuning job (default: None).
```
Comment (iaroslav-ai) [inline src/sagemaker/tuner.py:262]: NIt - it is unspecified for `Grid` strategy, and is `1` for other strategies.

File: src/sagemaker/tuner.py [hunk 2]
```diff
@@ -1351,7 +1356,7 @@ def create(
             objective_type (str): The type of the objective metric for evaluating training jobs.
                 This value can be either 'Minimize' or 'Maximize' (default: 'Maximize').
             max_jobs (int): Maximum total number of training jobs to start for the hyperparameter
-                tuning job (default: 1).
+                tuning job (default: None).
```
Comment (iaroslav-ai) [inline src/sagemaker/tuner.py:1359]: Same nit here on None / 1 being default.

**Review by iaroslav-ai — APPROVED** [review]
LGTM

**Review by claytonparnell — APPROVED** [review]

**Review by claytonparnell — COMMENTED** [review]
/bot run all

**Review by claytonparnell — COMMENTED** [review]
/bot run all

**Review by claytonparnell — COMMENTED** [review]
/bot run all

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

**Pull request merged**